commands/utils.go

```diff
@@ -128,6 +132,45 @@ type Result struct {
 
 type outputFunc func(testbedi.Core) (testbedi.Output, error)
 
+func mapListWithOutput(ranges [][]int, nodes []testbedi.Core, fns []outputFunc) ([]Result, error) {
+	var wg sync.WaitGroup
+	var lk sync.Mutex
+	var errs []error
+
+	total := 0
+	offsets := make([]int, len(ranges))
+	for i, list := range ranges {
+		offsets[i] = total
+		total += len(list)
+	}
+	results := make([]Result, total)
+
+	for i, list := range ranges {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			results_i, err := mapWithOutput(list, nodes, fns[i])
```

Comment:
I'm pretty sure we want to pass `list` and `i` in as parameters. I'm pretty sure this is a race issue. On small workloads it probably won't show up, but when things actually start to get schedule the value of i or list may have changed.

Reply:
Also something we need to make clear in our plugin documentation (none existed at the moment) is that they need to be thread safe, as now with this format the same node could be access at the same time.

Reply:
Ah, right, this is my first time using goroutines and I'm still getting used to stuff like this :+1:

Reply:
And gotcha on the documentation, I can throw that in when I do this rest of the documentation as well.

Reply:
Oh, just realized you said that there isn't any plugin documentation at all. I'll at least make an issue to remind us.